### PR TITLE
meta: update build cache dependency exclusions

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Currently changing the `.mocharc.jsonc` file causes the cache of the build to differ therefore rebuilding the entire project.
This PR works to exclude any non-build dependencies so that the build cache is not impacted by test/dev changes.

## Todos

- [x] Currently the entire `test` directory and subdirectory are excluded, are there any other files the need excluding?

### CODE REVIEW

**Comment by WikiRik** [comment]
We should do the same for the eslintrc and README files I think

**Comment by lohart13** [comment]
Eslint and markdown files have been excluded. What about the `typedoc.json`?

**Comment by WikiRik** [comment]
Yes, typedoc can also be excluded

**Comment by lohart13** [comment]
Rather than doing exclusions, I've changed to inclusion.

**Review by WikiRik — APPROVED** [review]

File: nx.json [hunk 1]
```diff
@@ -2,7 +2,11 @@
   "$schema": "node_modules/nx/schemas/nx-schema.json",
   "namedInputs": {
     "default": ["{projectRoot}/**/*"],
-    "prod": ["!{projectRoot}/test/**/*"]
+    "prod": [
+      "{projectRoot}/src/**/*",
+      "{projectRoot}/package.json",
+      "{projectRoot}/tsconfig.json"
```
Comment (ephys) [inline nx.json:8]: The root tsconfig extends the one from the test folder, but that one is not listed

(yes it's a weird setup, we'll be able to clean it up now that we only support TS 4.9+)

Comment (WikiRik) [reply nx.json:8]: The projectRoot here is the package folder. That tsconfig does not extend the one from the test folder right?

Comment (ephys) [reply nx.json:8]: Ah in that case, good to go

**Pull request merged**